public/app/features/integrated-alerting/components/AlertRules/AddAlertRuleModal/AddAlertRuleModal.test.tsx

```diff
@@ -18,31 +17,97 @@ jest.mock('app/core/app_events', () => {
+    threshold: '1 %',
+    lastNotified: '',
+    disabled: false,
+    rawValues: {
+      channels: [],
+      filters: [],
+      disabled: false,
+      template: {
+        name: 'pmm_mongodb_connections_memory_usage',
+        summary: 'Memory used by MongoDB connections',
+        params: [],
+      },
+      rule_id: '/rule_id/ded33d30-1b65-4b43-ba45-75ca52b48fa5',
+      summary: 'Just a summary',
+      params: [],
+      for: '3600s',
+      severity: 'SEVERITY_CRITICAL',
+      created_at: '2021-01-19T12:53:16.082610Z',
+    },
+  };
+
   afterEach(() => {
     jest.clearAllMocks();
   });
 
-  it('doesn not render the modal when visible is set to false', async () => {
-    let wrapper: ReactWrapper<{}, {}, any>;
+  it('should correctly render modal', () => {
+    const wrapper = mount(<AddAlertRuleModal setVisible={jest.fn()} isVisible />);
 
-    await act(async () => {
-      wrapper = mount(<AddAlertRuleModal setVisible={jest.fn()} isVisible={false} />);
-    });
+    expect(wrapper.find(dataQa('add-alert-rule-modal')).exists()).toBeTruthy();
+    expect(wrapper.find(dataQa('add-alert-rule-modal-form')).exists()).toBeTruthy();
+    expect(wrapper.find(dataQa('add-alert-rule-modal-add-button')).exists()).toBeTruthy();
+  });
 
-    expect(wrapper.find(dataQa('add-alert-rule-modal-form')).length).toBe(0);
+  it('does not render the modal when visible is set to false', async () => {
+    const wrapper = mount(<AddAlertRuleModal setVisible={jest.fn()} isVisible={false} />);
 
-    wrapper.unmount();
+    expect(wrapper.find(dataQa('add-alert-rule-modal-form')).length).toBe(0);
   });
 
   it('renders the modal when visible is set to true', async () => {
-    let wrapper: ReactWrapper<{}, {}, any>;
+    const wrapper = mount(<AddAlertRuleModal setVisible={jest.fn()} isVisible />);
+
+    expect(wrapper.find(dataQa('add-alert-rule-modal-form')).length).toBe(1);
+  });
 
-    await act(async () => {
-      wrapper = mount(<AddAlertRuleModal setVisible={jest.fn()} isVisible />);
+  it('should have the submit button disabled by default when adding a new rule', () => {
+    const wrapper = mount(<AddAlertRuleModal setVisible={jest.fn()} isVisible />);
+    const button = wrapper.find(dataQa('add-alert-rule-modal-add-button')).find('button');
+
+    expect(button.getDOMNode().hasAttribute('disabled')).toBeTruthy();
+  });
+
+  it('should enable the submit button if all fields are valid', () => {
+    const wrapper = mount(<AddAlertRuleModal setVisible={jest.fn()} isVisible alertRule={initialValues} />);
+
+    const thresholdInput = wrapper.find(dataQa('threshold-text-input'));
+    thresholdInput.simulate('change', {
+      target: {
+        value: '2',
+      },
     });
+    const button = wrapper.find(dataQa('add-alert-rule-modal-add-button')).find('button');
 
-    expect(wrapper.find(dataQa('add-alert-rule-modal-form')).length).toBe(1);
+    expect(button.getDOMNode().hasAttribute('disabled')).toBeFalsy();
```

Comment:
`getDOMNode().hasAttribute()` seems a bit verbose, why not use `.props('disabled')`? Would it work?

Reply:
nice one 👍